redash/serializers/query_result.py

```diff
@@ -0,0 +1,92 @@
+import cStringIO
+import csv
+import xlsxwriter
+from dateutil.parser import parse as parse_date
+from redash.utils import json_loads, UnicodeWriter
+from redash.query_runner import (TYPE_BOOLEAN, TYPE_DATE, TYPE_DATETIME)
+from redash.authentication.org_resolving import current_org
+
+
+def convert_format(fmt):
+    return fmt.replace('DD', '%d').replace('MM', '%m').replace('YYYY', '%Y').replace('YY', '%y').replace('HH', '%H').replace('mm', '%M').replace('ss', '%s')
+
+
+def serialize_query_result_to_csv(query_result):
+    s = cStringIO.StringIO()
+
+    query_data = json_loads(query_result.data)
+
+    fieldnames = []
+    bool_columns = []
+    date_columns = []
+    datetime_columns = []
+
+    for col in query_data['columns']:
+        fieldnames.append(col['name'])
+        if col['type'] == TYPE_BOOLEAN:

```

Comment:
These feel pretty mutually exclusive, so an `elif` might be in place.

Reply:
https://github.com/getredash/redash/pull/3841/commits/f26adedb9431c18e18bf1ac901eacc89489775d1